src/impl_constructors.rs

```diff
@@ -171,6 +171,19 @@ where
         }
         eye
     }
+
+    /// Create a 2D matrix from its diagonal
+    pub fn from_diag(diag: &ArrayBase<S, Ix1>) -> Self
```

Comment:
`from_diag` should be generic over the storage type of `diag`, so that you could e.g. create an `Array2` from a `ArrayView1` diagonal.

Reply:
I wasn't sure what I need to change to make the above comment possible.

Reply:
I was just talking about adding the `S2` type parameter, which you've already done.